### PR TITLE
Transaction, overview UI & code cleanup

### PR DESCRIPTION
**Pull request by oshorefueled**
This PR resolves issue #344. 

It also:
* contains code clean ups which include the addition of components.go, util.go and depricated.go file. Appropriate
   comments were added to each file describing the code that they contain. 
* modifies the progress bar widget. The default progress bar sets the max height to 4DP which doesn't allow a 8DP 
  height as specified in the overview mockup.
* adds a separator widget eliminating the need to initialize a Line object when initializing a page. And also repeatedly
  setting the width of the line. The same code is also present in PR #330.
* adds a Container type which is simply a wrapper to the layout.Inset type to help differentiate when it is used as 
   padding or margin. Its purpose is to help visualize the structure of a layout from the code.
* makes the transaction row layout reusable by both the transactions and overview page.

### CODE REVIEW

**Comment by JustinBeBoy** [comment]
- The right margin is not aligned


- The button wrong text when wallet status is offline

File: ui/decredmaterial/progressbar.go [hunk 1]
```diff
@@ -3,13 +3,97 @@
...
+	}
+
+	progressBarWidth := float32(gtx.Constraints.Max.X)
+	return layout.Stack{Alignment: layout.W}.Layout(gtx,
+		layout.Stacked(func(gtx layout.Context) layout.Dimensions {
+			return shader(progressBarWidth, p.TrackColor)
+		}),
+		layout.Stacked(func(gtx layout.Context) layout.Dimensions {
+			fillWidth := progressBarWidth * clamp1(p.Progress)
+			fillColor := p.Color
```
Comment (JustinBeBoy) [inline ui/decredmaterial/progressbar.go:56]: The progress color is "41BE53" and background is E6EAED

**Review by JustinBeBoy — CHANGES_REQUESTED** [review]

**Comment by JustinBeBoy** [comment]
The margin is 24

File: ui/decredmaterial/progressbar.go (hunk 1 above)
Comment (oshorefueled) [reply ui/decredmaterial/progressbar.go:56]: Yes, the colours for the track and progress are set on the pages using the progress bar widget

**Comment by oshorefueled** [comment]
> * The button wrong text when wallet status is offline
> 
> 

I'm not sure how to reproduce that error, you can create that as an issue and it will be resolved on another PR.

**Review by dreacot — APPROVED** [review]

**Review by dreacot — APPROVED** [review]

**Review by beansgum — APPROVED** [review]

**Review by JustinBeBoy — APPROVED** [review]

**Pull request merged**